dockstore-webservice/src/main/resources/migrations.1.13.0.xml

```diff
@@ -63,4 +63,19 @@
             ALTER TABLE user_profile ADD CONSTRAINT non_blank_link CHECK (CASE WHEN link IS NOT NULL then character_length(link) > 0 END);
         </sql>
     </changeSet>
+    <changeSet id="max_workflowname_and_toolname" author="coverbeck">
+        <!-- 256 is from StringInputValidationHelper.ENTRY_NAME_LENGTH_LIMIT.
+             If truncating, add numbers using a sequence to ensure workflowname uniqueness within a sourcecontrol, organization, repository;
+             otherwise a uniqueness constraint will cause a failure.
+        -->
+        <sql dbms="postgresql">
+            create sequence max_workflowname_and_toolname;
+            update workflow set workflowname = concat(left(workflowname, 254), nextval('max_workflowname_and_toolname')) where length(workflowname) > 256;
```

Comment:
if I understand correctly, this numbers everything that gets truncated, not just the ones with duplicates correct?
that said, probably doesn't matter much because we don't have anything legit above 254 chars?

Reply:
> if I understand correctly, this numbers everything that gets truncated, not just the ones with duplicates correct? that said, probably doesn't matter much because we don't have anything legit above 254 chars?

Yes, our "legit" longest workflowname is 57 chars. Only dev currently has entries with workflowname greater than than 256 chars, and all of those have the same first 256 characters (they're about 5 of them).

Since we're changing the name regardless and breaking references to the workflows anyway, and since I'd find it hard to write SQL that only adds the numbers for dupes, I figured keep it simple since it's only going to affect dev data anyway.

Reply:
This code might limit the number of truncated names to 99 (when the ID reaches 100, the generated truncated+numbered name is longer than 256 chars).  Is there ever a case where that matters in practice?  I can't think of one off the top of my head...

Reply:
There are only 8 workflows in dev with names > 256, and there's already a fix deployed to prod and dev that prevents additional ones from being created.